### ISSUE TITLE
Gulp.src/gulp.dest does not copy images or fonts correctly in gulp@5.0.0

### ISSUE THREAD

**Issue by marcelotena** [gulpjs/gulp#2796]
### What were you expecting to happen?

Wrote a task to optimize images using gulp-imagemin and move compressed files to a destination folder.

### What actually happened?

Files did not compress and moved files could not be opened. I can see that it also happens when fonts are moved using gulp.src/gulp.dest, resulting files can't be opened.

Everything works correctly if I roll back to version 4.0.2.

### Please give us a sample of your gulpfile

Here's a link to my gulpfile: https://github.com/marcelotena/gulp-frontend-starter/blob/main/gulpfile.babel.js

-->
```js
import gulp from 'gulp'
import {default as imagemin, gifsicle, mozjpeg, optipng} from 'gulp-imagemin'

gulp.task('imgmin', () => {
    return gulp.src('./src/img/*')
        .pipe(imagemin([
            gifsicle({interlaced: true}),
            mozjpeg({quality: 60, progressive: true}),
            optipng({optimizationLevel: 5})
        ]))
        .pipe(gulp.dest('./dist/img'));
});
```

### Terminal output / screenshots

No issues are displayed in the terminal.

### Please provide the following information:
* OS & version: macOS 12.2.1
* node version (run `node -v`): 21.7.3
* npm version (run `npm -v`): 10.5.0
* gulp version (run `gulp -v`): 5.0.0

**Comment by Sebazzz** [comment]
Also happens with a simple `gulp.src('./node_modules/material-symbols/*.woff2').pipe(gulp.dest('./wwwroot/build/fonts/'))`. 

The file sizes are different, it appears that the files are incorrectly read or copied:

![image](https://github.com/gulpjs/gulp/assets/1426097/e269c8d1-d6ed-48bf-9526-f23ef78f4a8c)

Seems to be some kind of encoding problem, or reading binaries in text mode:

![image](https://github.com/gulpjs/gulp/assets/1426097/f7119876-5e05-4d6b-b92e-64a0811ab769)

**Comment by restless** [comment]
Please see: https://github.com/gulpjs/gulp/issues/2790

TL;DR
try: `.src(path, {encoding: false})

**Comment by marcelotena** [comment]
Thank you @restless, setting up encoding to false solves the issue.
